src/LowerTriangularMatrices/lower_triangular_matrix.jl

```diff
@@ -59,59 +75,92 @@ only the lower triangle (the non-zero entries) of `L`."""
+@inline Base.getindex(L::LowerTriangularArray, r::AbstractRange, I...) = getindex(L.data, r, I...)
+@inline Base.getindex(L::LowerTriangularArray, i::Integer) = getindex(L.data,i)
+
+Base.@propagate_inbounds function Base.setindex!(L::LowerTriangularArray{T,N}, x, I::Vararg{Any, M}) where {T,N,M} 
+    @boundscheck M == N-1 || throw(BoundsError(L, I))
+    setindex!(L.data, x, I...)
+end 
 
-Base.@propagate_inbounds Base.setindex!(L::LowerTriangularMatrix, x, k::Integer) = setindex!(L.data, x, k)
-Base.@propagate_inbounds function Base.setindex!(L::LowerTriangularMatrix{T}, x, i::Integer, j::Integer) where T
+Base.@propagate_inbounds function Base.setindex!(L::LowerTriangularArray{T,N}, x, I::Vararg{Any, N}) where {T,N}
+    i, j = I[1:2] # TODO: check if i,j are Int?
     @boundscheck i >= j || throw(BoundsError(L, (i, j)))
     k = ij2k(i, j, L.m)
-    setindex!(L.data, x, k)
+    setindex!(L.data, x, k, I[3:end]...)
 end
 
-@inline Base.setindex!(L::LowerTriangularMatrix, x::AbstractVector, r::AbstractRange) = setindex!(L.data, x, r)
+@inline Base.setindex!(L::LowerTriangularMatrix, x::AbstractVector, r::AbstractRange, I...) = setindex!(L.data, x, r, I...)
 
 # propagate index to data vector
-Base.eachindex(L ::LowerTriangularMatrix)    = eachindex(L.data)
-Base.eachindex(Ls::LowerTriangularMatrix...) = eachindex((L.data for L in Ls)...)
+Base.eachindex(L ::LowerTriangularArray)    = eachindex(L.data)
+Base.eachindex(Ls::LowerTriangularArray...) = eachindex((L.data for L in Ls)...)
 
 """
-    unit_range = eachharmonic(L::LowerTriangular)
+    unit_range = eachharmonic(L::LowerTriangularArray)
 
-creates `unit_range::UnitRange` to loop over all non-zeros in a LowerTriangularMatrix `L`.
+creates `unit_range::UnitRange` to loop over all non-zeros/spherical harmonics numbers in a LowerTriangularArray `L`.
 Like `eachindex` but skips the upper triangle with zeros in `L`."""
-eachharmonic(L::LowerTriangularMatrix) = eachindex(L.data)
+eachharmonic(L::LowerTriangularArray) = axes(L.data, 1) # TODO: do we want to keep this like that for n-dim arrays or change it? also adjust docstring when this is decided 
 
 """
     unit_range = eachharmonic(Ls::LowerTriangularMatrix...)
 
 creates `unit_range::UnitRange` to loop over all non-zeros in the LowerTriangularMatrices
 provided as arguments. Checks bounds first. All LowerTriangularMatrix's need to be of the same size.
 Like `eachindex` but skips the upper triangle with zeros in `L`."""
-eachharmonic(Ls::LowerTriangularMatrix...) = eachindex(Ls...)
+function eachharmonic(Ls::LowerTriangularArray...) 
+    N_harmonics = [size(L.data,1) for L in Ls]
+    @boundscheck all(y->y==N_harmonics[1], N_harmonics) || throw(BoundsError)
+    return eachharmonic(Ls[1]) # TODO: do we want to keep this like that for n-dim arrays or change it? also adjust docstring when this is decided 
+end 
 
 # CONVERSIONS
 """ 
     L = LowerTriangularMatrix(M)
```

Comment:
While we're at it you can always write
```julia
"""
$(TYPEDSIGNATURES)
This function does something cool."""
abc(x,y) = x*y
```
And the signature is inserted automatically, similar for structs with `$(TYPEDFIELDS)`